documentation/modules/ROOT/pages/connectors/mongodb-sink.adoc

```diff
@@ -170,6 +171,133 @@ Replace the placeholders in the example with the actual values for your environm
+Appropriate data type mapping ensures that the original data is accurately represented in the destination database.
+
+The {prodname} MongoDB sink connector resolves a column's data type by using a logical or primitive type-mapping system.
+Primitive types include values such as integers, floating points, Booleans, strings, and bytes.
+Typically, Kafka messages use a specific Kafka Connect `Schema` type code to represent primitive data types.
+
+By contrast, to represent more complex data, {prodname} uses logical schema names, which provide logical groupings of named fields that can represent a range of data types (strings, arrays, JSON, XML, and so forth).
+These structured types take into account the semantic meaning of the data and provide a more opinionated interpretation about how to serialize the underlying primitive types.
+Logical types are useful in representing values that have a specific encoding, such as numbers that represent the time since the epoch.
+
+For a JSON string, JSON map, or a raw BSON byte array, the connector uses the default data type mappings from MongoDB to store the document.
+
+The following examples show representative structures of primitive and logical data types:
+
+.Primitive field schema
+====
+[source,json]
+----
+{
+  "schema": {
+    "type": "INT64"
+  }
+}
+----
+====
+
+.Logical field schema
+====
+[source,json]
+----
+{
+  "schema": {
+    "type": "INT64",
+    "name": "org.apache.kafka.connect.data.Date"
+  }
+}
+----
+====
+
+
+Kafka Connect is not the only source for these complex logical types.
+For example, when a {prodname} source connector emits a change event, it assigns similar logical types to event fields to represent such data types as timestamps, dates, and even JSON data.
+
+
+The {prodname} MongoDB sink connector uses these primitive and logical types to resolve a column's type to a MongoDB BSON document field type.
+The following tables illustrate the primitive and logical mappings between Kafka Connect and MongoDB BSON types, and between {prodname} and MongoDB BSON types.
+
+. xref:mongodb-sink-kafka-connect-primitive-mappings[]
+. xref:mongodb-sink-kafka-connect-logical-mappings[]
+
+
+// Type: reference
+// Title: Mappings between Kafka Connect primitive data types and MongoDB field types
+// ModuleID: debezium-mongodb-sink-connector-mappings-between-kc-primitive-data-types-and-mongodb-field-types
+[[mongodb-sink-kafka-connect-primitive-mappings]]
+
+.Mappings between Kafka Connect primitive data types and MongoDB field data types
+|===
+|Primitive Type |MongoDB BSON Type
+
+|INT8, INT16, INT32
+|BsonInt32
+
+|INT64
+|BsonInt64
+
+|FLOAT32, FLOAT64
+|BsonDouble
+
+|BOOLEAN
+|BsonBoolean
+
+|STRING
+|BsonString
+
+|BYTES
+|BsonBinary
+
+|===
+
+
+// Type: reference
+// Title: Mappings between Kafka Connect logical data types and MongoDB field types
+// ModuleID: debezium-mongodb-connector-mappings-betw-kc-logical-data-types-and-mongodb-field-types
```

Comment:
```suggestion

```
Remove these annotations for the Nebel tooling if we're dispensing with the topic headings.